src/main/java/frc/robot/commands/SwerveControllerFollower.java

```diff
@@ -0,0 +1,102 @@
+import frc.robot.subsystems.Swerve;
+
+/**
+ * Custom implementation of
+ * {@link edu.wpi.first.wpilibj2.command.SwerveControllerCommand
+ * SwerveControllerCommand}
+ * to simplify construction, allow PathPlanner paths, and enable custom logging
+ */
+public class SwerveControllerFollower extends CommandBase {
+
+    private final Swerve drivetrain;
+    private final Trajectory trajectory;
+    private Timer timer = new Timer();
+    private Pose2d initialPathPlannerPose = null;
+
+    public SwerveControllerFollower(Swerve drivetrain, Trajectory trajectory) {
+        this.drivetrain = drivetrain;
+        this.trajectory = trajectory;
+
+        addRequirements(drivetrain);
+    }
+
+    public SwerveControllerFollower(Swerve drivetrain, String pathFilename, TrajectoryConfig config,
+            boolean isReversed, boolean isInitial) {
+        this.drivetrain = drivetrain;
+        addRequirements(drivetrain);
+
+        PathPlannerTrajectory ppTrajectory = PathPlanner.loadPath(pathFilename, config.getMaxVelocity(),
+                config.getMaxAcceleration(), isReversed);
+        this.trajectory = ppTrajectory;
+
+        if (isInitial) {
+            initialPathPlannerPose = new Pose2d(
+                    ppTrajectory.getInitialPose().getTranslation(),
+                    ppTrajectory.getInitialState().holonomicRotation);
+        }
+    }
+
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        if (initialPathPlannerPose != null) {
+            drivetrain.resetOdometry(initialPathPlannerPose);
+        }
+
+        timer.reset();
+        timer.start();
+        // drivetrain.resetPathController(); // reset theta setpoint between different
+        // trajectories
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+        double currentTime = timer.get();
+        // The target state of the trajectory right now (the robot's pose and velocity)
+        Trajectory.State targetState = trajectory.sample(currentTime);
+        Rotation2d targetRotation = targetState.poseMeters.getRotation();
+        // Check if we are using PathPlanner trajectories
+        if (trajectory instanceof PathPlannerTrajectory) {
+            targetState = ((PathPlannerTrajectory) trajectory).sample(currentTime);
+            targetRotation = ((PathPlannerState) targetState).holonomicRotation;
+        }
+
+        drivetrain.drive(targetState, targetRotation, true);
+
+        // Pose2d targetPose = targetState.poseMeters;
+        // SmartDashboard.putNumber("Target Heading",
+        // targetPose.getRotation().getDegrees());
+        // SmartDashboard.putNumber("Target X", targetPose.getX());
+        // SmartDashboard.putNumber("Target Y", targetPose.getY());
+    }
+
+    // Called once the command ends or is interrupted.
+    @Override
+    public void end(boolean interrupted) {
+        timer.stop();
```

Comment:
If this command is interrupted, won't the drivetrain still be moving (possibly rapidly) from the last call to `drivetrain.drive()`? Feels like you wanna tell the drivetrain to stop if the command is interrupted.

Reply:
fixed!